src/reconcile/rec_child.c

```diff
@@ -114,6 +127,7 @@ __rec_child_deleted(
     if (!visible_all) {
         cmsp->del = *page_del;
         cmsp->state = WT_CHILD_PROXY;
+        page_del->on_disk = true;
```

Comment:
It is fine to set this flag here (since it just means we decided to write the deletion out, not that we necessarily did)... but, since at this point the reconciliation could still fail and not actually write anything, the name should probably be something else. Maybe "selected_for_ds" or "selected_for_write"?

Reply:
Thanks, used `selected_for_write` as a variable name.